File: nim/sublist/sublist.nim
import sequtils

type Result* = enum
  Unequal, Equal, Sublist, SuperList

proc contains[T](x, y: openArray[T]): bool =
  anyIt(0 .. x.len - y.len, y == x[it ..< it + y.len])

proc sublist*(a, b: openArray[int]): Result =
  if a == b: result = Equal
  elif a in b: result = Sublist
  elif b in a: result = Superlist

File: dart/sublist/lib/sublist.dart
enum Classification { equal, unequal, sublist, superlist }

class Sublist {

  Classification sublist(List lst1, List lst2) {
    String s1 = lst1.join(',') + ',';
    String s2 = lst2.join(',') + ',';
    return s1 == s2
        ? Classification.equal
        : s1.contains(s2)
            ? Classification.superlist
            : s2.contains(s1)
                ? Classification.sublist
                : Classification.unequal;
  }
}
